### ISSUE TITLE
Kotlin @Suppress("UNCHECKED_CAST") causes 'CHECKCAST kotlin/Unit' bytecode problem

### ISSUE THREAD

**Issue by lgou2w** [lgou2w/ldk#82]
See line 248:

https://github.com/lgou2w/ldk/blob/3918d3600a08270919fbe8bfaa88d179c3f5b12f/ldk-nbt/src/main/kotlin/com/lgou2w/ldk/nbt/NBT.kt#L239-L255

After reviewing the bytecode, there will be an extra `CHECKCAST kotlin/Unit` directive under the `remove(key)` function.

See below:
```java
L14
  LINENUMBER 252 L14
  ALOAD 0
  ALOAD 1
  INVOKEVIRTUAL com/lgou2w/ldk/nbt/NBTTagCompound.remove (Ljava/lang/Object;)Ljava/lang/Object;
  CHECKCAST kotlin/Unit // <--- HERE
  POP
  GOTO L15
L13
L16
  LINENUMBER 251 L16
L15
  LINENUMBER 249 L15
L12
  NOP
L17
```

It throws an `java.lang.ClassCastException: com.lgou2w.ldk.nbt.NBTTagShort cannot be cast to kotlin.Unit` error when I debug with the following code:

```kotlin
val compound = ofCompound { putShort("key", 1) }
  .removeIf<Short>("key") { it == 1.toShort() }
```

When I remove `@Suppress("UNCHECKED_CAST")`, the check command disappears, I think this is a problem with the kotlin compiler itself. But I will fix a version urgently and remove it. 😢

**Comment by lgou2w** [comment]
I'm not sure if the kotlin problem tracker has the same problem, but I'm sure this is the kotlin compiler's own problem.

**Comment by lgou2w** [comment]
Fixed in `0.1.8-beta12-hotfix2` .
